### DISCUSSION TITLE
Ruff project roadmap?

### DISCUSSION THREAD

**Discussion by qarmin** [astral-sh/ruff#9629] [General]
I'm really interested in these two ideas
- vulture alternative(dead code finder) - https://github.com/astral-sh/ruff/issues/872
- mypy alternative(type checker) - https://github.com/astral-sh/ruff/issues/3893

However, from what I can see from the changes in subsequent releases, the priority is probably to stabilize version 1.0

Is it already known in what direction ruff will be developed and what are the estimated times of the above tools?(if, of course, there will be attempts to implement them)

**Comment by charliermarsh** [comment]
I believe we will attempt to do both of these, though we haven't started on either in earnest. Some of the larger projects we're queueing up will indeed be related to stability, rule categorization, and overall user experience. I'm intentionally not kicking off the type checker yet since we have a lot of other commitments and there's a lot of research that needs to go into it.

Are you interested in working on dead-code detection? We had a very nice but partial implementation of a control-flow graph (https://github.com/astral-sh/ruff/pull/5384) that wasn't finished. I'd love to support someone in picking that up, extending it to support try-catch, and starting our dead-code analysis by adding unreachable coed detection within function definitions.

**Comment by mikeleppane** [comment]
+1 for dead code analysis. This is one thing I miss from the JetBrains products which handle this quite well.
I would be interested to start looking into this and as a start implementing dead code analysis for function defs.
